rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1278,3 +1280,80 @@ def get_basic_contract_info(self, address: ChecksumEvmAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def _update_curve_decoder(self, tx_decoder: 'EVMTransactionDecoder') -> None:
+        try:
+            curve_decoder = tx_decoder.decoders['Curve']
+        except KeyError as e:
+            raise InputError(
+                'Expected to find Curve decoder but it was not loaded. '
+                'Please open an issue on github.com/rotki/rotki/issues if you saw this.',
+            ) from e
+        new_mappings = curve_decoder.reload()
+        tx_decoder.address_mappings.update(new_mappings)
+
+    def curve_protocol_cache_is_queried(
+            self,
+            tx_decoder: Optional['EVMTransactionDecoder'],
+    ) -> bool:
+        """
+        Make sure that information that needs to be queried is queried and if not query it
+        Returns true if the cache was modified or false otherwise.
+        If the tx_decoder provide is None no information for the decoders is reloaded
+
+        Updates curve pools cache.
+        1. Deletes all previous cache values
+        2. Queries information about curve pools' addresses, lp tokens and used coins
+        3. Saves queried information in the cache in globaldb
+        """
+        # Do a query in the local chache to avoid a trip to the database every time
+        last_update_ts, decoder_updated = self.protocols_information_queried_at.get(GeneralCacheType.CURVE_LP_TOKENS, (Timestamp(0), False))  # noqa: E501
```

Comment:
This is not needed in my opinion.

This function seems to really not be called often. A "trip" to the database is nothing. The DB is our cache for most cases.

I would only add an in-memory cache for something in addition to the DB cache if it's queried extremely often and then would also try to measure to see how much sense it makes.

Reply:
Okey, probably I'm over optimizing/engineering here. It was because if you have `k` curve pool tokens you are doing `2k-1` queries to the database but is a small case and it might not be that bad